jax/_src/numpy/ufuncs.py

```diff
@@ -152,15 +152,25 @@ def _arccosh(x: ArrayLike, /) -> Array:
     out = _where(real(out) < 0, lax.neg(out), out)
   return out
 
+def _preserving_dtype_floor(x: ArrayLike, /) -> Array:
+    if dtypes.isdtype(x.dtype, "integral"):
+      return x
```

Comment:
@jakevdp what would be the best way to transform an ArrayLike to Array ?

Reply:
`return lax.asarray(x)`